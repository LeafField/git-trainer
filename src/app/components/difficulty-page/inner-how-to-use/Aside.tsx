import React, { FC, useEffect } from "react";

type Props = {
  isHidden: boolean;
};

const Aside: FC<Props> = ({ isHidden }) => {
  const [isMobile, setIsMobile] = React.useState<boolean>(false);

  useEffect(() => {
    const resize = new ResizeObserver((entries) => {
      entries.forEach((entry) => {
        if (entry.contentRect.width < 1024) {
          setIsMobile(true);
        } else {
          setIsMobile(false);
        }
      });
    });
    resize.observe(document.body);

    return () => {
      resize.disconnect();
    };
  }, []);

  return (
    <aside
      className="absolute inset-0 col-start-1 col-end-2 row-start-2 row-end-3 bg-consoleHeader px-4 pt-[6rem] text-white transition-all
       duration-300 ease-in-out aria-hidden:invisible  aria-hidden:opacity-0 lg:visible lg:static lg:bg-inherit lg:px-0 lg:pt-0"
      aria-hidden={isMobile ? `${isHidden}` : "false"}
      id="aside"
      data-testid="aside"
    >
      <h3 className="text-lg lg:text-2xl">フォルダ名、リモート名などの注意</h3>
      <p className="pt-9">
        リモートリポジトリのURLは「URL」と打ってください
        <br />
        例:git remote add origin URL
      </p>
      <p className="pt-4">
        フォルダ名を指定する場合は「.」と打ってください
        <br />
        例:git add .
      </p>
    </aside>
  );
};

export default Aside;
