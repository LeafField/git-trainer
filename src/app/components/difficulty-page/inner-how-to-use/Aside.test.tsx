import { act, render, screen } from "@testing-library/react";
import Aside from "./Aside";
import { mockResizeObserver } from "jsdom-testing-mocks";

const resizeObserver = mockResizeObserver();

describe("Asideの単体テスト", () => {
  it("Asideのタイトルが正しく表示されているか", () => {
    render(<Aside isHidden={false} />);
    expect(
      screen.getByText("フォルダ名、リモート名などの注意")
    ).toBeInTheDocument();
  });

  it("画面サイズの変更に対し、aria-hiddenのstateが正しく管理されているか", () => {
    render(<Aside isHidden={true} />);
    resizeObserver.mockElementSize(document.body, {
      contentBoxSize: { inlineSize: 1024, blockSize: 768 },
    });
    act(() => {
      resizeObserver.resize(document.body);
    });
    expect(screen.getByTestId("aside")).toHaveAttribute("aria-hidden", "false");

    resizeObserver.mockElementSize(document.body, {
      contentBoxSize: { inlineSize: 1023, blockSize: 768 },
    });
    act(() => {
      resizeObserver.resize(document.body);
    });
    expect(screen.getByTestId("aside")).toHaveAttribute("aria-hidden", "true");
  });

  it("Asideのスナップショットテスト", () => {
    const { container } = render(<Aside isHidden={false} />);
    expect(container).toMatchSnapshot();
  });
});
