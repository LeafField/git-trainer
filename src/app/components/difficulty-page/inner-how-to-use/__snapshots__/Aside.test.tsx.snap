// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asideの単体テスト Asideのスナップショットテスト 1`] = `
<div>
  <aside
    aria-hidden="false"
    class="absolute inset-0 col-start-1 col-end-2 row-start-2 row-end-3 bg-consoleHeader px-4 pt-[6rem] text-white transition-all duration-300 ease-in-out aria-hidden:invisible  aria-hidden:opacity-0 lg:visible lg:static lg:bg-inherit lg:px-0 lg:pt-0"
    data-testid="aside"
    id="aside"
  >
    <h3
      class="text-lg lg:text-2xl"
    >
      フォルダ名、リモート名などの注意
    </h3>
    <p
      class="pt-9"
    >
      リモートリポジトリのURLは「URL」と打ってください
      <br />
      例:git remote add origin URL
    </p>
    <p
      class="pt-4"
    >
      フォルダ名を指定する場合は「.」と打ってください
      <br />
      例:git add .
    </p>
    <p
      class="mt-4"
    >
      「-m」等のオプションは打たないでください
      <br />
      例:git commit
    </p>
    <p
      class="mt-4"
    >
      現在、コマンドの判定がシビアになっています。
      <br />
      コマンドの半角スペースは一個のみ、コマンドの最後にスペースが入らないように注意してください。
    </p>
  </aside>
</div>
`;
